Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -97,6 +97,23 @@ public class AlertStore {
             }
         }
     }
+
+    public func recordRetractedAlert(_ alert: Alert, at date: Date, completion: ((Result<Void, Error>) -> Void)? = nil) {
+        self.managedObjectContext.performAndWait {
+            let storedAlert = StoredAlert(from: alert, context: self.managedObjectContext, issuedDate: date)
+            storedAlert.retractedDate = date
+            do {
+                try self.managedObjectContext.save()
+                self.log.default("Recorded alert: %{public}@", alert.identifier.value)
```

Comment:
[nit] consider logging "Recorded _retracted_ alert" (just to be clearer & more specific)